workflows/mapping/Snakefile

```diff
@@ -0,0 +1,266 @@
+# -*- snakemake -*-
+
+import yaml
+import itertools
+import pandas
+
+try:
+    config.update(yaml.load(open('config.yaml')))
+except IOError:
+    raise IOError('No "config.yaml" found; please supply another config '
+                  ' file using the --configfile argument')
+
+
+# ----------------------------------------------------------------------------
+# Helper functions
+#
+
+
+# srcdir() is a snakemake built-in function. WRAPPERS must be relative to this
+# file.
+HERE = srcdir('')
+WRAPPERS = '../../wrappers'
+def wrapper_for(tool):
+    return os.path.join(HERE, WRAPPERS, tool)
+
+
+def params_for(rule, key):
+    d = config.get('rules', {}).get(rule, {}).get('params', {}).get(key, '')
+    return d
+
+
+def threads_for(rule):
+    return config.get('rules', {}).get(rule, {}).get('threads', {})
+
```

Comment:
I was thinking the same. I wasn't sure where such a module should live, and how exactly it should be imported. `sys.path` hackery would be the easiest solution I guess.